File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Request/Filter/Element/CoreFilterRequestElementInterface.js
import {BaseModel} from "../../../../../../UIFrameWork/Shared/Common/BaseModel.js";

export default class CoreFilterRequestElementInterface extends BaseModel {
    constructor() {
        super();
    }

    setDataModelRegistry(dataModelRegistry) {
        this.dataModelRegistry = dataModelRegistry;
    }

    getDataModelRegistry() {
        return this.dataModelRegistry;
    }
}

CoreFilterRequestElementInterface.Register = () => {
    return Object.freeze({
        registerKey_ElementWithOperand: Symbol("ElementWithOperand"),
        registerKey_ElementWithOperationParamValue: Symbol("ElementWithOperationParamValue"),
    });
}

CoreFilterRequestElementInterface.Method = () => {
    return Object.freeze({
        dataModelRegistryPropName: Symbol("dataModelRegistry"),
    });
}

File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/request/filter/element/CoreFilterRequestElementInterface.java
package org.infra.reactive.form.engine.form.engine.model.dto.request.filter.element;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.infra.reactive.form.engine.form.engine.model.dto.request.filter.element.jackson.CoreFilterRequestElementInterfaceDeserialize;

@JsonDeserialize(using = CoreFilterRequestElementInterfaceDeserialize.class)
public interface CoreFilterRequestElementInterface {
    public static final String dataModelRegistryPropName = "dataModelRegistry";
    public static final String registerKey_ElementWithOperand = "ElementWithOperand";
    public static final String registerKey_ElementWithOperationParamValue = "ElementWithOperationParamValue";
    String getDataModelRegistry();
}
